airflow/models/taskreschedule.py

```diff
@@ -75,7 +75,7 @@ class TaskReschedule(TaskInstanceDependencies):
         ),
     )
     dag_run = relationship("DagRun")
-    execution_date = association_proxy("dag_run", "execution_date")
```

Comment:
Another thing we should remove in a future PR (this is for compatibility)

Reply:
Keeping it for now as discussed.